espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/MotionEventBuilder.java

```diff
@@ -62,48 +81,112 @@ public MotionEventBuilder setDeviceId(int deviceId) {
         return this;
     }
 
+    public MotionEventBuilder setSource(int source) {
+        params.source = source;
+        return this;
+    }
+
     public MotionEventBuilder setEdgeFlags(int edgeFlags) {
         params.edgeFlags = edgeFlags;
         return this;
     }
 
-    public void run () throws AppiumException {
+    public MotionEventBuilder setPointerType(PointerType pointerType) {
+        params.pointerType = pointerType;
+        return this;
+    }
+
+    public MotionEvent run () throws AppiumException {
```

Comment:
nullable

Reply:
I don't think the MotionEvent should ever be null

Reply:
there's `return null` below